src/lib/message/L2ToL1Message.ts

```diff
@@ -72,32 +72,28 @@ export type L2ToL1MessageReaderOrWriter<T extends SignerOrProvider> =
 export class L2ToL1Message {
   public static fromEvent<T extends SignerOrProvider>(
     l1SignerOrProvider: T,
-    event?: L2ToL1TransactionEvent['args'],
-    outboxAddress?: string,
-    batchNumber?: BigNumber,
-    indexInBatch?: BigNumber
+    event: L2ToL1TransactionEvent['args'],
+    outboxAddress: string,
+    isNitro: boolean
```

Comment:
instead of having this be user provided, can't we just look at `event.arbBlockNum` and check if its nitro?

Reply:
we'd need to make a network call to find that arb block, and this is a constructor so it's only sync.

Reply:
Alternatively we could change the signature of `public static fromEvent` to make it async

Reply:
ye we can make fromEvent async and grab the info